ui/selectors/approvals.ts

```diff
@@ -1,19 +1,35 @@
 import { ApprovalControllerState } from '@metamask/approval-controller';
 import { ApprovalType } from '@metamask/controller-utils';
+import { TransactionMeta } from '../../shared/constants/transaction';
 
 type ApprovalsMetaMaskState = {
   metamask: {
     pendingApprovals: ApprovalControllerState['pendingApprovals'];
+    unapprovedTxs: {
+      [transactionId: string]: TransactionMeta;
+    };
   };
 };
 
-export function hasPendingApprovalsSelector(
+export const getApprovalRequestsByType = (
   state: ApprovalsMetaMaskState,
   approvalType: ApprovalType,
-) {
+) => {
   const pendingApprovalRequests = Object.values(
     state.metamask.pendingApprovals,
   ).filter(({ type }) => type === approvalType);
 
+  return pendingApprovalRequests;
+};
+
+export function hasPendingApprovalsSelector(
```

Comment:
Should we remove the `Selector` here also?

Reply:
Fixed